ttnn/cpp/ttnn/operations/data_movement/pad/pad.cpp

```diff
@@ -80,16 +80,18 @@ static ttnn::Tensor pad_impl(
 
     ShapeType output_padded_shape;
     for(size_t i = 0; i < padding.size(); i++) {
-        output_padded_shape[i] = input_shape_with_tile_padding[i] + padding[i].second;
+        output_padded_shape[i] = padding[i].first + input_shape_with_tile_padding[i] + padding[i].second;
     }
 
     auto pad_front = padding | std::views::transform([](const auto& p) { return p.first; });
     auto pad_back = padding | std::views::transform([](const auto& p) { return p.second; });
 
     const bool front_padding_is_zero = std::accumulate(pad_front.begin(), pad_front.end(), 0) == 0;
-    TT_FATAL(
-        front_padding_is_zero,
-        "ttnn.pad: on device padding does not support front padding");
+    if (input_tensor.get_layout() == ttnn::TILE_LAYOUT) {
+        TT_FATAL(
```

Comment:
Where is the test for this?

Reply:
it's in tests/ttnn/unit_tests/operations/test_pad.py, I modified the existing test, for both front padding and end padding.